### PR TITLE
Move lessons to electives folder

### LINKED ISSUE #1256: Update elective lessons and migrate all to /electives folder
All lesson files listed in the root [README as electives](https://github.com/Techtonica/curriculum#optional--elective-lessons) move those files into a new `/electives` folder.
~SEO: Search Engine Optimization~
- [x] Sass
- [x] Python
~Datetime~
- [x] Data Visualization
- [x] Adding Chai/Mocha Tests to your project
- [x] MongoDB
- [x] Mongoose
- [x] Auth0
- [x] Memoization
- [x] Mobile Development
In addition, add the following files to the elective section in the README and also put them into the /electives folder:

~/agile-development (move the entire folder and all its lessons)~
~/deploying/deploying.md~
- [x] /web-patterns/web-patterns.md
- [x] /databases/database-migrations.md
- [x] /backend/1_intro_to_backend.md
- [x] /optimization/optimization.md

For all moved files, please update all links that exist in the curriculum already to point to their new location. If you any links that are absolute, e.g. use https://github... then convert them to relative links.

### PR DESCRIPTION
**Pull request by FrozenGirl-spd**
Absolute links were converted to relative links.
Extra files were removed
Required files were moved into electives folder
Updated README.md

#fixes #1256 

@alodahl

### CODE REVIEW

**Comment by Kundan28** [comment]
And please resolve the conflicts.

**Comment by Kundan28** [comment]
@FrozenGirl-spd You have 4 days left, if we don't get any response from your side then the issue will be assigned to someone else.

**Comment by FrozenGirl-spd** [comment]
Can you check it out?

**Comment by Kundan28** [comment]
> Can you check it out?

Please resolve the conflicts too.

**Comment by FrozenGirl-spd** [comment]
How to do it? It says "Only those with write access to this repository" can.

**Comment by Kundan28** [comment]
> How to do it? It says "Only those with write access to this repository" can.

Follow these steps @FrozenGirl-spd 
image

**Comment by FrozenGirl-spd** [comment]
I followed these steps.. 
#1346

**Comment by Kundan28** [comment]
> I followed these steps..
> #1346

You should never create a new for resolving the conflicts, the conflicts should be resolved under same opened PR.

**Comment by FrozenGirl-spd** [comment]
Yep okiee..... sorry..

File: README.md [hunk 1]
```diff
@@ -165,7 +165,7 @@ Apprentices should memorize [**this**](Cheatsheet.md) within the first month.
 1. Coding assessment
 1. App development assessment
 1. [Recursion](/recursion/recursion.md)
-1. [Asset Optimization](/optimization/optimization.md)
+1. [Asset Optimization](/electives/optimization.md)
```
Comment (alodahl) [inline README.md:168]: Should this be moved to the elective section if it is an elective now, @vegetabill ?

Comment (alodahl) [reply README.md:168]: Hi @FrozenGirl-spd , just move this to the electives section and you'll be ready for merge.

File: oauth/authorization.md [hunk 2]
```diff
@@ -13,7 +13,7 @@ About 1.5 hour
 
 Here are links to lessons that should be completed before this lesson:
```
Comment (alodahl) [inline oauth/authorization.md:15]: @vegetabill should this be an elective if it has an elective prereq?

**Comment by alodahl** [comment]
Hi @FrozenGirl-spd , @vegetabill and I just had a meeting about electives. Could you please return these 4 to _not_ being electives:
- datetime
- seo
- Agile
- Deployment

Thanks for understanding!

@vegetabill I'm going to add our meeting notes to https://github.com/Techtonica/curriculum/issues/1256.

**Comment by FrozenGirl-spd** [comment]
Please review... and suggest me.. @alodahl  @Kundan28  @vegetabill

File: optimization/optimizing-your-react-node-project.md [hunk 3]
```diff
@@ -8,7 +8,7 @@ About 1.5 hours
 
 Here are links to lessons that should be completed before this lesson:
 
-- [Intro to Optimization](/optimization/optimization.md)
```
Comment (alodahl) [inline optimization/optimizing-your-react-node-project.md:11]: @vegetabill also looking for your opinion on this one... is this still a prereq?

File: README.md [hunk 4]
```diff
@@ -231,24 +231,27 @@ These are not blockers for any of the core sequence and so they are worked into
...
+1. [Whiteboarding Interview Challenges](/interviewing/week-3-whiteboard-challenges.md)
 1. [Agile Project Management](/agile-development/intro-to-agile.md)
 1. [Intro to Scrum](/agile-development/scrum.md)
 
 ### Optional / Elective Lessons
 
-- [SEO: Search Engine Optimization](/seo/seo.md)
-- [Sass](/sass/sass.md)
-- [Python](/python/python.md)
-- [Datetime](/datetime/datetime.md)
```
Comment (alodahl) [inline README.md:244]: Thank you for fixing these.  Can you please move this Datetime line to line 105?

File: README.md [hunk 5]
```diff
@@ -145,7 +145,7 @@ Apprentices should memorize [**this**](Cheatsheet.md) within the first month.
 1. [Relational Databases](/databases/relational-databases.md)
 1. [Table Relationships and Join Queries by Treehouse](https://teamtreehouse.com/library/querying-relational-databases)
 1. [Adding Tests to your NodeJS project](/testing-and-tdd/integration-testing.md)
-1. [Database Migrations](/databases/database-migrations.md)
+1. [Database Migrations](/electives/databases/database-migrations.md)
```
Comment (alodahl) [inline README.md:148]: Please move all new electives to the electives section below.

File: README.md (hunk 4 above)
Comment (FrozenGirl-spd) [reply README.md:244]: should i replace line 105 with datetime line or just add?

File: README.md [hunk 6]
```diff
@@ -102,7 +102,7 @@ Apprentices should memorize [**this**](Cheatsheet.md) within the first month.
 1. [JavaScript 3 - Conditionals, Comparisons, Booleans](/javascript/javascript-3-conditionals.md)
 1. [JavaScript 4 - Loops](/javascript/javascript-4-loops.md)
 1. [JavaScript 5 - Switch Statements](/javascript/javascript-5-switch.md)
-1. [JavaScript 6 - Object Literals](/javascript/javascript-6-object-literals.md)
+1. [Datetime](/datetime/datetime.md)
```
Comment (Kundan28) [inline README.md:105]: Please don't remove that JavaScript 6 file. Just add the Datetime file just after that line. @FrozenGirl-spd

File: README.md (hunk 4 above)
Comment (alodahl) [reply README.md:244]: just add

File: README.md (hunk 5 above)
Comment (alodahl) [reply README.md:149]: If anything has just become an elective, move it out of its present section and into the Elective section below.

File: oauth/authorization.md (hunk 2 above)
Comment (vegetabill) [reply oauth/authorization.md:15]: OAuth is an elective

File: optimization/optimizing-your-react-node-project.md (hunk 3 above)
Comment (vegetabill) [reply optimization/optimizing-your-react-node-project.md:11]: Any "optimizing/optimization" lesson is elective, so both this and its prereq

Comment (alodahl) [reply optimization/optimizing-your-react-node-project.md:11]: @frozenGirl-spd please rename this file, and move this link to electives.

File: oauth/authorization.md (hunk 2 above)
Comment (alodahl) [reply oauth/authorization.md:15]: @FrozenGirl-spd change this one too please and move the link to electives

**Comment by alodahl** [comment]
@FrozenGirl-spd please address these comments so we can merge:

https://github.com/Techtonica/curriculum/pull/1326#discussion_r426199501
https://github.com/Techtonica/curriculum/pull/1326#discussion_r426199346
https://github.com/Techtonica/curriculum/pull/1326#discussion_r419773545
https://github.com/Techtonica/curriculum/pull/1326#discussion_r422530241

**Comment by alodahl** [comment]
@FrozenGirl-spd , will you be able to make these changes before GSSoC ends?

File: README.md [hunk 7]
```diff
@@ -231,24 +232,27 @@ These are not blockers for any of the core sequence and so they are worked into
 1. [Diversity, Inclusion & Implicit Bias](/diversity-inclusion-bias/inclusion.md)
 1. [Tech Tips and Rules](https://docs.google.com/document/d/1SWiUl0lYQ0sYDM8vQnVhJVOvT4MRA-9pDglKOgqTWak/edit)
 1. [Giving Presentations](giving-presentations/giving-presentations.md)
-1. [Whiteboarding](/career/interviewing/whiteboarding.md)
-1. [Whiteboarding Interview Challenges](/career/interviewing/week-3-whiteboard-challenges.md)
+1. [Whiteboarding](/interviewing/whiteboarding.md)
```
Comment (alodahl) [inline README.md:235]: why did you change these?  The files are definitely still in the career/ folder.

File: README.md [hunk 8]
```diff
@@ -165,7 +166,7 @@ Apprentices should memorize [**this**](Cheatsheet.md) within the first month.
 1. Coding assessment
 1. App development assessment
 1. [Recursion](/recursion/recursion.md)
-1. [Asset Optimization](/optimization/optimization.md)
+1. [Asset Optimization](/electives/optimization.md)
```
Comment (alodahl) [inline README.md:169]: move this to electives

File: optimization/optimizing-your-react-node-project.md (hunk 3 above)
Comment (alodahl) [reply optimization/optimizing-your-react-node-project.md:11]: sorry, I meant keep this and change it to the new link, and make sure it is moved to the electives section in the Readme.

File: projects/mongo-assessment.md [hunk 9]
```diff
@@ -24,9 +24,8 @@ In this project, you will have 30 minutes to create a new database and a few doc
 3. Email your required screenshots to be reviewed at the end of the assessment.
 
 ## Setup Instructions
-
-- [ ] Know how to take screenshots on your machine. For mac, you can press `command + shift + 4` to draw a box around the part of the screen you want to capture (in this case, your command line terminal), and it should appear on your desktop with the name "Screenshot..." followed by a hash.
-- [ ] You should already have MongoDB installed after [this lesson](/Users/alodahl/Projects/curriculum/databases/mongo-db.md).
+- [ ] Know how to take screenshots on your machine.  For mac, you can press `command + shift + 4` to draw a box around the part of the screen you want to capture (in this case, your command line terminal), and it should appear on your desktop with the name "Screenshot..." followed by a hash.
+- [ ] You should already have MongoDB installed after [this lesson](/Users/alodahl/Projects/curriculum/electives/databases/mongo-db.md).
```
Comment (alodahl) [inline projects/mongo-assessment.md:28]: this is clearly my fault, but could you change this from an alodahl URL to the real github elective URL?

File: README.md [hunk 10]
```diff
@@ -80,7 +80,7 @@ Apprentices should memorize [**this**](Cheatsheet.md) within the first month.
 1. [Recipe Page Project: CSS](/projects/recipe-page/phase-2-css-prompt.md)
 1. [Media Queries](/web/media-queries.md)
 1. [UI/UX](/ui-ux-design/ui-ux-design.md)
-1. [Web patterns](/web-patterns/web-patterns.md)
+1. [Web patterns](/electives/web-patterns.md)
```
Comment (alodahl) [inline README.md:83]: move this to electives

File: optimization/optimizing-your-react-node-project.md (hunk 3 above)
Comment (FrozenGirl-spd) [reply optimization/optimizing-your-react-node-project.md:11]: move or copy it to elective section in readme?

Comment (alodahl) [reply optimization/optimizing-your-react-node-project.md:11]: copy

Comment (alodahl) [reply optimization/optimizing-your-react-node-project.md:11]: please also bring optimization/optimizing-your-react-node-project.md into electives, fix any links to the new path, and put it in the right place in the readme.

File: projects/mongo-assessment.md [hunk 11]
```diff
@@ -25,7 +25,7 @@ In this project, you will have 30 minutes to create a new database and a few doc
 
 ## Setup Instructions
 - [ ] Know how to take screenshots on your machine.  For mac, you can press `command + shift + 4` to draw a box around the part of the screen you want to capture (in this case, your command line terminal), and it should appear on your desktop with the name "Screenshot..." followed by a hash.
-- [ ] You should already have MongoDB installed after [this lesson](/Users/alodahl/Projects/curriculum/electives/databases/mongo-db.md).
+- [ ] You should already have MongoDB installed after [this lesson](/github/Techtonica/curriculum/electives/databases/mongo-db.md).
```
Comment (alodahl) [inline projects/mongo-assessment.md:28]: This new link doesn't work.

File: README.md [hunk 12]
```diff
@@ -238,17 +236,22 @@ These are not blockers for any of the core sequence and so they are worked into
...
+- [Mongoose](/electives/databases/mongoose.md)
+- [Auth0](/electives/oauth/o-auth.md)
+- [Memoization](/electives/memoization.md)
+- [Mobile Development](/electives/mobile-development.md)
+- [web-patterns](/electives/web-patterns.md)
+- [Database Migrations](/electives/databases/database-migrations.md)
+- [backend](/electives/1_intro_to_backend.md)
+- [Optimization](/electives/optimization.md)
+- [angular](/electives/angular.md)
+- [Asset Optimization](/electives/optimization.md)
```
Comment (alodahl) [inline README.md:253]:
```suggestion
- [Asset Optimization](/electives/optimization.md)
- 
```
What is going on with line 251, 253, and 254? Are they the same file?
are these really going to be called /electives/optimization.md and /electives/optimization/optimization.md? Please make sure they are named correctly and there are no repeats.

Comment (alodahl) [reply README.md:253]: Also, where is "optimizing-your-react-node-project.md"?

**Comment by Kundan28** [comment]
@FrozenGirl-spd Are you still working on this ? Please complete it before 31st May so that we have sufficient time to review it.

File: README.md (hunk 12 above)
Comment (FrozenGirl-spd) [reply README.md:253]: As you mentioned earlier, optimization.md is an elective. Is "optimizing-your-react-node-project.md" also an elective?

Comment (FrozenGirl-spd) [reply README.md:253]: I have just moved ../optimization.md/optimization.md to electives section.

**Comment by FrozenGirl-spd** [comment]
@alodahl  @Kundan28  I have pushed the required changes.

**Comment by Kundan28** [comment]
@FrozenGirl-spd Resolve the conflicts too

**Comment by Kundan28** [comment]
Updated the scoresheet.
